src/components/KSelect/KSelect.vue

```diff
@@ -2,278 +2,195 @@
         >
-          <div
-            v-if="appearance === 'button'"
-            :id="selectInputId"
-            class="k-select-button"
-            data-testid="k-select-input"
-            role="listbox"
-            style="position: relative;"
+          <KInput
+            :id="selectId"
+            v-bind="modifiedAttrs"
+            autocapitalize="off"
+            autocomplete="off"
+            class="select-input"
+            :class="{ 'filtering-disabled': !enableFiltering,
+                      'hide-model-value': hasCustomSelectedItem
+            }"
+            data-testid="select-input"
+            :disabled="isDisabled"
+            :error="error"
+            :help="help"
+            :label="label ? strippedLabel : undefined"
+            :label-attributes="labelAttributes"
+            :model-value="filterQuery"
+            :placeholder="selectedItem && !enableFiltering ? selectedItem.label : placeholderText"
+            :readonly="isReadonly"
+            @blur="onInputBlur"
+            @focus="onInputFocus"
+            @keypress="onInputKeypress"
+            @keyup="(evt: any) => triggerFocus(evt, isToggled)"
+            @update:model-value="onQueryChange"
           >
-            <KButton
-              :id="selectTextId"
-              v-bind="modifiedAttrs"
-              appearance="tertiary"
-              :style="widthStyle"
-              @keyup="(evt: any) => triggerFocus(evt, isToggled)"
+            <template #after>
+              <CloseIcon
+                v-if="isClearVisible"
+                data-testid="clear-selection-icon"
+                role="button"
+                tabindex="0"
+                @click="clearSelection"
+              />
+              <ChevronDownIcon
+                class="chevron-down-icon"
+                :class="{ 'disabled': isDisabled }"
+              />
+            </template>
+            <template
+              v-if="$slots['label-tooltip']"
+              #label-tooltip
+            >
+              <slot name="label-tooltip" />
+            </template>
+          </KInput>
+          <Transition name="kongponents-fade-transition">
+            <div
+              v-if="hasCustomSelectedItem && (!enableFiltering || !isToggled.value)"
+              class="custom-selected-item-wrapper"
+              :class="{ 'clearable': clearable }"
             >
               <slot
                 :item="selectedItem"
                 name="selected-item-template"
               >
-                {{ selectButtonText }}
-                <ChevronDownIcon />
+                <slot
+                  :item="selectedItem"
+                  name="item-template"
+                />
               </slot>
-            </KButton>
+            </div>
+          </Transition>
+        </div>
+        <template #content>
+          <div
+            v-if="enableFiltering && loading"
+            class="select-loading"
+            data-propagate-clicks="false"
```

Comment:
question: what is `data-propagate-clicks`?

Reply:
Sometimes clicks on containers are propagated up which results in unintentional clicks on other elements. I added this data property to control that when we can't use `pointer-events: none;`.

Reply:
can you document this in the comments somewhere?

Reply:
Added a comment in `onSelectWrapperClick` method.